recipe/install_nvcc.sh

```diff
@@ -2,61 +2,98 @@
 
 set -xeuo pipefail
 
-# Default to using `nvcc` to specify `CUDA_HOME`.
-if [ -z ${CUDA_HOME+x} ]
-then
-    CUDA_HOME="$(dirname $(dirname $(which nvcc)))"
-fi
-
 # Set `CUDA_HOME` in an activation script.
 mkdir -p "${PREFIX}/etc/conda/activate.d"
-cat > "${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
+cat >"${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
 #!/bin/bash
 
-if [[ ! -d "${CUDA_HOME}" ]]
+export CUDA_HOME_CONDA_NVCC_BACKUP="\${CUDA_HOME:-}"
+export CFLAGS_CONDA_NVCC_BACKUP="\${CFLAGS:-}"
+export CPPFLAGS_CONDA_NVCC_BACKUP="\${CPPFLAGS:-}"
+export CXXFLAGS_CONDA_NVCC_BACKUP="\${CXXFLAGS:-}"
+
+# Default to using \$(cuda-gdb) to specify \$(CUDA_HOME).
```

Comment:
Why should we use `cuda-gdb` to do this (as opposed to `nvcc` or something else)?

Reply:
Because once we activate the environment there's a new `nvcc` higher in the path which is our shim script to call into the actual `nvcc` 😄

Reply:
> Because once we activate the environment there's a new `nvcc` higher in the path which is our shim script to call into the actual `nvcc` 😄

Right! 🤦‍♂